base/abstractarray.jl

```diff
@@ -83,6 +83,20 @@ function axes(A)
     map(OneTo, size(A))
 end
 
+"""
+    is_non1_indexed(A)
+    is_non1_indexed(A, B, ...)
+
+Return `true` if the indices of `A` start with something other than 1 along any axis.
+If multiple arguments are passed, equivalent to `is_non1_indexed(A) | is_non1_indexed(B) | ...`.
+
+This defaults to false for an arbitrary object.
+"""
+is_non1_indexed(obj)                 = false    # this is not a required trait
```

Comment:
I think I'd be okay with calling this `has_offset_axes` and then requiring `axes` to be defined.  It's still not defined for Tuples, but I say we rectify that.  Everything else that supports broadcasting needs to support axes, so it doesn't seem to be that big of a requirement.

Reply:
It's not just Tuples, it's also `SuiteSparse.Sparse` and a few other similar "opaque" objects. I guess we could define `axes` for them too, since even `<:Factorization` has `axes` defined.

But this raises the thought that going with `is_offset_indexed` might be better, depending on whether we think indexing or `axes` is the more "fundamental" operation.